src/ray/core_worker/core_worker.cc

```diff
@@ -238,6 +249,24 @@ void CoreWorkerProcess::RunTaskExecutionLoop() {
   instance_.reset();
 }
 
+void CoreWorkerProcess::RunStatsService() {
+  absl::MutexLock lock(&CoreWorkerProcess::stats_mutex_);
```

Comment:
I can't understand why the mutex helps. You can assume that `CoreWorkerProcess::Initialize` is thread-safe (although it's not currently), so the mutex here is unnecessary.

Reply:
Um.. I put stats initialization in core worker constructor, so it's used for make sure init exactly once. But it could be removed.